### PR TITLE
ConsoleJanitor: add all discord loggers

### PR DESCRIPTION
**Pull request by sadan4**
adds all discord loggers

adds an exception for routing utils

### CODE REVIEW

File: src/plugins/consoleJanitor/index.ts [hunk 1]
```diff
@@ -103,20 +120,14 @@ export default definePlugin({
...
-            find: logger,
+        {
+            find: "Σ:",
             predicate: () => settings.store.disableNoisyLoggers,
-            all: true,
             replacement: {
-                match: new RegExp(String.raw`new \i\.\i${logger.replace(/([()])/g, "\\$1")}`),
-                replace: `$self.NoopLogger${logger}`
+                match: /(&&)(console)/,
+                replace: "$1$self.shouldLog(arguments[0])&&$2"
```
Comment (Nuckyz) [inline src/plugins/consoleJanitor/index.ts:128]:
```suggestion
                match: /(?<=&&)(?=console)/,
                replace: "$self.shouldLog(arguments[0])&&"
```

File: src/plugins/consoleJanitor/index.ts [hunk 2]
```diff
@@ -34,15 +34,32 @@ const settings = definePluginSettings({
...
 export default definePlugin({
     name: "ConsoleJanitor",
     description: "Disables annoying console messages/errors",
-    authors: [Devs.Nuckyz],
+    authors: [Devs.Nuckyz, Devs.sadan],
     settings,
 
+    start(){
+        if(!settings.store.enableNavigationRouter)
+            logAllow.add("Routing/Utils");
```
Comment (Nuckyz) [inline src/plugins/consoleJanitor/index.ts:60]:
```suggestion
        if (!settings.store.enableNavigationRouter) {
            logAllow.add("Routing/Utils");
        }
```

File: src/plugins/consoleJanitor/index.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ const NoopLogger = {
     time: Noop,
     fileOnly: Noop
 };
-
+const logAllow = new Set(["Spotify"]);
```
Comment (Nuckyz) [inline src/plugins/consoleJanitor/index.ts:24]:
```suggestion

const logAllow = new Set(["Spotify"]);

```

**Review by Nuckyz — CHANGES_REQUESTED** [review]
disableSpotifyLogger is useless now. Also I think maybe just make an option to whitelist the loggers listed, then add Routing/Utils and GatewaySocket as default

**Comment by sadan4** [comment]
> disableSpotifyLogger is useless now. Also I think maybe just make an option to whitelist the loggers listed, then add Routing/Utils and GatewaySocket as default

Like a text field where the user can write loggers to whitelist?

**Comment by Nuckyz** [comment]
> > disableSpotifyLogger is useless now. Also I think maybe just make an option to whitelist the loggers listed, then add Routing/Utils and GatewaySocket as default
> 
> Like a text field where the user can write loggers to whitelist?

yes

**Comment by sadan4** [comment]
cant be bothered to fix git issue
will make new pr

**Pull request closed without merging**